R/foreign-keys.R

```diff
@@ -84,6 +84,21 @@ dm_add_fk <- function(dm, table, columns, ref_table, ref_columns = NULL, ...,
     ref_col_name <- get_key_cols(ref_key)
   } else {
     ref_col_name <- names(eval_select_indices(ref_col_expr, colnames(ref_table_obj)))
+    # check if either a PK or UK already matches ref_col_name
+    all_keys <- dm_get_all_pks_impl(dm) %>%
+      rename(uk_col = pk_col) %>%
+      bind_rows(dm_get_all_uks_impl(dm)) %>%
+      filter(table == ref_table_name)
+    # setequal() could also be used for matching, but IMHO the order should matter
+    matches_keys <- map_lgl(all_keys$uk_col, identical, ref_col_name)
+    if (!any(matches_keys)) {
+      if (check) {
+        if (!is_unique_key_se(ref_table_obj, ref_col_name)$unique) {
```

Comment:
Where is `is_unique_key_se()` defined?

Reply:
https://github.com/cynkra/dm/blob/main/R/key-helpers.R#L76